jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/MultiPart.java

```diff
@@ -267,18 +275,21 @@ public void writeTo(Path path) throws IOException
                     IO.copy(Content.Source.asInputStream(newContentSource()), out);
                 }
                 this.path = path;
-                this.temporary = false;
             }
             else
             {
                 this.path = Files.move(this.path, path, StandardCopyOption.REPLACE_EXISTING);
+            }
+
+            try (AutoLock ignored = lock.lock())
+            {
                 this.temporary = false;
             }
         }
 
         public void delete() throws IOException
         {
-            if (this.path != null)
+            if (this.path != null && Files.exists(this.path))
```

Comment:
`Files.exist()` seems redundant.
If the implementation sets a value for `path` we know it exists; if it is set by the application, it deserves `Files.delete()` to throw.

Reply:
I've done the change to set `this.path` to null after deletion, but this has one side-effect: calling `getPath()` after `delete()` now returns null while it previously returned the deleted path.